rasa/nlu/selectors/response_selector.py

```diff
@@ -290,10 +318,127 @@ def process(self, message: Message, **kwargs: Any) -> None:
         )
 
-        prediction_dict = {"response": label, "ranking": label_ranking}
+        prediction_dict = {
+            "response": label,
+            "ranking": label_ranking,
+            "full_intent_name": full_intent_name,
+        }
 
         self._set_message_property(message, prediction_dict, selector_key)
 
+    def persist(self, file_name: Text, model_dir: Text) -> Dict[Text, Any]:
+        """Persist this model into the passed directory.
+
+        Return the metadata necessary to load the model again.
+        """
+
+        if self.model is None:
+            return {"file": None}
+
+        model_dir = Path(model_dir)
+        tf_model_file = model_dir / f"{file_name}.tf_model"
+
+        io_utils.create_directory_for_file(tf_model_file)
+
+        self.model.save(str(tf_model_file))
+
+        io_utils.pickle_dump(
+            model_dir / f"{file_name}.data_example.pkl", self.data_example
+        )
+        io_utils.pickle_dump(
+            model_dir / f"{file_name}.label_data.pkl", self._label_data
+        )
+        io_utils.json_pickle(
+            model_dir / f"{file_name}.index_label_id_mapping.pkl",
+            self.index_label_id_mapping,
+        )
+        io_utils.json_pickle(
+            model_dir / f"{file_name}.index_tag_id_mapping.pkl",
+            self.index_tag_id_mapping,
+        )
+        io_utils.json_pickle(
+            model_dir / f"{file_name}.retrieval_intent_mapping.pkl",
+            self.retrieval_intent_mapping,
+        )
+        return {"file": file_name}
+
+    @classmethod
+    def load(
+        cls,
+        meta: Dict[Text, Any],
+        model_dir: Text = None,
+        model_metadata: Metadata = None,
+        cached_component: Optional["DIETClassifier"] = None,
+        **kwargs: Any,
+    ) -> "DIETClassifier":
+        """Loads the trained model from the provided directory."""
+
+        if not model_dir or not meta.get("file"):
+            logger.debug(
+                f"Failed to load model. "
+                f"Maybe the path '{os.path.abspath(model_dir)}' doesn't exist?"
+            )
+            return cls(component_config=meta)
+
+        (
+            index_label_id_mapping,
+            index_tag_id_mapping,
+            label_data,
+            meta,
+            data_example,
+            retrieval_intent_mapping,
+        ) = cls._load_from_files(meta, model_dir)
+
+        meta = train_utils.update_similarity_type(meta)
+
+        model = cls._load_model(
+            index_tag_id_mapping, label_data, meta, data_example, model_dir
+        )
+
+        return cls(
+            component_config=meta,
+            index_label_id_mapping=index_label_id_mapping,
+            index_tag_id_mapping=index_tag_id_mapping,
+            model=model,
+            retrieval_intent_mapping=retrieval_intent_mapping,
+        )
+
+    @classmethod
+    def _load_from_files(cls, meta: Dict[Text, Any], model_dir: Text):
```

Comment:
I would not overwrite this method to avoid code duplication. Just add
```
        retrieval_intent_mapping = io_utils.json_unpickle(
            model_dir / f"{file_name}.retrieval_intent_mapping.pkl"
        )
```
to the `load` method.